src/se/terhol/pisemka22/AirplaneImpl.java

```diff
@@ -0,0 +1,83 @@
+package se.terhol.pisemka22;
+
+/**
+ * @author Tereza Holm
+ */
+public class AirplaneImpl implements Airplane {
+    Passenger[] seatList;
+    int capacity;
+
+    public AirplaneImpl(int capacity) {
+        if (capacity <= 0) {
+            throw new IndexOutOfBoundsException();
+        }
+        this.capacity = capacity;
+        this.seatList = new PassengerImpl[capacity];
+
+
+    }
+
+    @Override
+    public Passenger getPassenger(int seat) {
+        if(seat < 0 || seat >= this.getCapacity()) {
+            throw new IndexOutOfBoundsException();
+        }
+        return seatList[seat];
+    }
+
+    @Override
+    public int getCapacity() {
+        return this.capacity;
+    }
+
+    @Override
+    public int getPassengerReservation(Passenger pas) throws ReservationException {
+        if(pas == null) {
+            throw new ReservationException();
+        }
+        int isFound = -1;
+        for(int i = 0; i<this.getCapacity(); i++) {
+            if (seatList[i] != null && seatList[i].equals(pas)) {
+                isFound = i;
+            }
+        }
+
+        if (isFound < 0) {
+            throw new ReservationException("No such passenger");
+        }
+        return isFound;
+    }
+
+    @Override
+    public void makeReservation(Passenger pas) throws ReservationException {
+
+        int freeSeat = -1;
+        for(int i = 0; i<this.getCapacity(); i++) {
+            if(this.seatList[i] == null){
+                freeSeat = i;
+                seatList[freeSeat] = pas;
+                return;
```

Comment:
It is better to not return from inside of cicle/middle of the methods. Could be easily overlooked by other programmers when working this code. We can use some private method maybe?

Reply:
Done